_posts/2017-05-09-contributing.md

```diff
@@ -0,0 +1,161 @@
+Development of Cytoscape.js requires several additional packages which are defined in `package.json` including Gulp (build/ test/ formatting/ etc. automation), ESLint (formatting), and Mocha and Chai (tests and assertions), among others.
+Install these by navigating to the folder containing Cytoscape.js and running `npm install`.
+Cytoscape.js comes with a `.eslint.json` file which ESLint will use when inspecting your code, helping to reduce bugs and maintain a uniform style.
+
+Starting with version 3.2.0, Cytoscape.js uses [Babel](https://babeljs.io/) for transpiling, so you can take advantage of ES2015/ ES6 features.
+
+#### Style
+
+On the topic of uniform style: the Cytoscape.js doesn't have a strict style guide.
+The only rules are to use two spaces for indentation and use single-quotes around strings; other than this, just try to match the style of existing code.
+
+Code can be automatically formatted by running `gulp format`; make sure that you've committed all changes before running this command so that formatting changes are kept separate from the underlying code changes.
+
+#### Committing
+
+As you write code, commit changes with `git add .` and `git commit -m "work on awesome feature"`.
+If using GitHub Desktop, [review the guide from GitHub](https://help.github.com/desktop/guides/contributing/committing-and-reviewing-changes-to-your-project/) about committing changes.
+
+> Frequency of commits is a personal preference; however, I recommend committing after completing each "unit of work" where each unit is a block of changes to one or more files that do not break compilation of Cytoscape.js.
+
+#### Code Organization
+
+Important folders for development are described here:
+
+- `src/`: The `src/` folder is likely where you'll spend most of your time; this folder contains the many files that are combined at build-time to create the `cytoscape.js` file that is used in applications.
+- `test/`: The `test/` folder contains tests for Cytoscape.js. If you are contributing a feature that can be tested, please add tests for this feature. If you are contributing a bugfix, add a test that fails without your bugfix.
+- `documentation/`: The `documentation/` folder contains documentation for all public features of Cytoscape.js (internal/ private APIs need not be documented here). For new features, please add documentation on how to use them here.
+- `dist/`: The `dist/` folder contains the most recent publicly-released version of Cytoscape.js. This is the version that is published to npm and [js.cytoscape.org/](http://js.cytoscape.org/) on each new release.
+- `build/`: Building Cytoscape.js will compile the many files within `src/` and emit two files here: one containing a non-minified build of Cytoscape.js and one containing a minified build.
+
+#### Building and Testing Cytoscape.js
+
+To build and test new features or bugfixes, run `gulp build` (to build Cytoscape.js without testing) or `gulp test` (to run all tests).
```

Comment:
npm scripts are there on master as well. They're absent on the 2.x branch, but do you think they should be added since 2.x is no longer the target for active development?